src/Nethermind/Nethermind.Blockchain/BeaconBlockRoot/BeaconBlockRootHandler.cs

```diff
@@ -1,39 +1,48 @@
 // SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
-using Nethermind.Core.Specs;
+using System;
 using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Crypto;
+using Nethermind.Evm.Tracing;
+using Nethermind.Evm.TransactionProcessing;
 using Nethermind.Int256;
-using Nethermind.State;
-using Nethermind.Core.Crypto;
-using Nethermind.Core.Extensions;
-
-namespace Nethermind.Consensus.BeaconBlockRoot;
 
-public class BeaconBlockRootHandler : IBeaconBlockRootHandler
+namespace Nethermind.Blockchain.BeaconBlockRoot;
+public class BeaconBlockRootHandler(ITransactionProcessor processor) : IBeaconBlockRootHandler
 {
-    public void ApplyContractStateChanges(Block block, IReleaseSpec spec, IWorldState stateProvider)
-    {
-        if (!spec.IsBeaconBlockRootAvailable ||
-            block.IsGenesis ||
-            block.Header.ParentBeaconBlockRoot is null)
-            return;
-
-        Address eip4788Account = spec.Eip4788ContractAddress ?? Eip4788Constants.BeaconRootsAddress;
+    private const long GasLimit = 30_000_000L;
```

Comment:
Should this be from chainspec, why this value?

Reply:
directly from EIP: https://eips.ethereum.org/EIPS/eip-4788